inst/tests/tests.Rraw

```diff
@@ -16796,37 +16795,24 @@ if (test_xts) {
     index = structure(c(1167782400, 1167868800), tzone = "UTC", tclass = "Date"),
     .Dim = c(2L, 6L), .Dimnames = list(NULL, c("SPY.Open", "SPY.High", "SPY.Low", "SPY.Close", "SPY.Volume", "SPY.Adjusted"))
   )
-  test(2108.35, first(xt), xt_first, output="using xts::first: is.xts(x)")
-  test(2108.36, first(xt, n=2L), xt_first2, output="using xts::first: is.xts(x)")
-} else {
-  test(2108.21, last(x, n=2L), 2:3, output="using utils::tail: !is.xts(x) & nargs>1 & !'package:xts'%in%search()")
-  test(2108.22, last(y, n=2L), y[2:3], output="using utils::tail: !is.xts(x) & nargs>1 & !'package:xts'%in%search()")
-  test(2108.23, last(x, n=1L), 3L, output="using utils::tail: !is.xts(x) & nargs>1 & !'package:xts'%in%search()")
-  test(2108.24, last(y, n=1L), y[3L], output="using utils::tail: !is.xts(x) & nargs>1 & !'package:xts'%in%search()")
-  test(2108.25, last(xt), error="you should have 'xts' installed already")
-  test(2108.26, last(xt, n=2L), error="you should have 'xts' installed already")
-  test(2108.31, first(x, n=2L), 1:2, output="using utils::head: !is.xts(x) & nargs>1 & !'package:xts'%in%search()")
-  test(2108.32, first(y, n=2L), y[1:2], output="using utils::head: !is.xts(x) & nargs>1 & !'package:xts'%in%search()")
-  test(2108.33, first(x, n=1L), 1L, output="using utils::head: !is.xts(x) & nargs>1 & !'package:xts'%in%search()")
-  test(2108.34, first(y, n=1L), y[1L], output="using utils::head: !is.xts(x) & nargs>1 & !'package:xts'%in%search()")
-  test(2108.35, first(xt), error="you should have 'xts' installed already")
-  test(2108.36, first(xt, n=2L), error="you should have 'xts' installed already")
+  test(2108.35, first(xt), xt_first, output="using xts::first")
+  test(2108.36, first(xt, n=2L), xt_first2, output="using xts::first")
 }
-test(2108.41, last(x), 3L, output="using 'x[[length(x)]]': !is.xts(x) & !nargs>1 & is.null(dim(x))")
-test(2108.42, last(y), y[3L], output="using 'x[[length(x)]]': !is.xts(x) & !nargs>1 & is.null(dim(x))")
-test(2108.51, first(x), 1L, output="using 'x[[1L]]': !is.xts(x) & !nargs>1 & is.null(dim(x))")
-test(2108.52, first(y), y[1L], output="using 'x[[1L]]': !is.xts(x) & !nargs>1 & is.null(dim(x))")
-test(2108.61, last(df), structure(list(a=2L, b=2L), row.names=2L, class="data.frame"), output="using 'x[nrow(x),]': !is.xts(x) & !nargs>1 & is.data.frame(x)")
-test(2108.62, last(dt), data.table(a=2L, b=2L), output="using 'x[nrow(x),]': !is.xts(x) & !nargs>1 & is.data.frame(x)")
-test(2108.71, first(df), structure(list(a=1L, b=3L), row.names=1L, class="data.frame"), output="using 'x[1L,]': !is.xts(x) & !nargs>1 & is.data.frame(x)")
-test(2108.72, first(dt), data.table(a=1L, b=3L), output="using 'x[1L,]': !is.xts(x) & !nargs>1 & is.data.frame(x)")
+test(2108.41, last(x), 3L)
+test(2108.42, last(y), y[3L])
+test(2108.51, first(x), 1L)
+test(2108.52, first(y), y[1L])
+test(2108.610, last(df), structure(list(a=2L, b=2L), row.names=2L, class="data.frame"))
```

Comment:
why can't we use data.frame() here? at least a comment seems warranted

Reply:
now uses data.frame()